subprojects/docs/src/samples/userguide/java/customDirs/build.gradle

```diff
@@ -0,0 +1,52 @@
+plugins {
+    id 'java'
+}
+
+sourceCompatibility = '1.8'
+targetCompatibility = '1.8'
+version = '1.2.1'
+
+// START SNIPPET custom-src-dirs
+sourceSets {
+    main {
+         java {
+            srcDirs = ['src']
+         }
+    }
+
+    test {
+        java {
+            srcDirs = ['test']
+        }
+    }
+}
+// END SNIPPET custom-src-dirs
+
+// START SNIPPET custom-extra-src-dir
+sourceSets {
+    main {
+        java {
+            srcDir 'thirdParty/src/main/java'
+        }
+    }
+}
+// END SNIPPET custom-extra-src-dir
+
+
+// START SNIPPET custom-source-set
+sourceSets {
+    intTest
+}
+// END SNIPPET custom-source-set
+
+// START SNIPPET custom-report-dirs
+reporting.baseDir = 'my-reports'
+testResultsDirName = 'my-test-results'
+
+task showDirs {
+    doLast {
+        println project.reportsDir
```

Comment:
Should use the `Logger` instance instead of `println`.

Reply:
I've been going by past practice in the manual, and `println()` is always guaranteed to print out to the console. Also note that it's used several times in the _Working with files_ chapter. I don't feel strongly about this as we're clearly using `println()` for demonstration purposes, but I also understand that if we always use `println()`, that's what users will do.

Reply:
I'd say the same logic applies here that is usually applied to application code. Use the logger instead of `System.out.println`. What I see a lot is that users simply copy/paste whatever they find and therefore apply practices that are less optimal. For this example I'd personally use `logger.quiet`.

We should probably make a full sweep through the whole user guide and replace `println` with logger usage. Created an issue here: https://github.com/gradle/gradle/issues/4905

Reply:
OK, I didn't know about the `quiet()` option, which appears to do the opposite of what I would expect. Wish it had been called `always()` or something like that. Quiet sounds like something that would be quiet, similar to trace. Anyway, I see the logic, but it's will take a little getting used to :smile: